File: spec/test_lexer_unknown_character.c
#include "testparser.h"
#include <assert.h>
#include <string.h>

int main()
{
    char const * input = "x";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_UNEXPECTED_INPUT);
    p_context_delete(context);

    input = "123";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    assert(p_result(context) == 123u);
    p_context_delete(context);

    return 0;
}

File: spec/test_lexer_unknown_character.rs
use testparser::*;

fn main() {
    let mut context = p_context_new(b"x");
    assert_eq!(P_UNEXPECTED_INPUT, p_parse(&mut context));
    p_context_delete(context);

    let mut context = p_context_new(b"123");
    assert_eq!(P_SUCCESS, p_parse(&mut context));
    assert_eq!(123, p_result(&context));
    p_context_delete(context);
}
